### DISCUSSION TITLE
Add a custom provider

### DISCUSSION THREAD

**Discussion by heiwais25** [nextauthjs/next-auth#742] [Help]
**Your question**
Where do I need to modify if I want to contribute to the project by adding a new custom provider?

**What are you trying to do**
Recently, I implemented a new custom provider for Naver (https://www.naver.com/) which is the major web site in South Korea. To contribute to the project, I added a new provider configuration in this path (`src/providers/naver.js`). The content of the file is like this. (https://github.com/heiwais25/next-auth)

```
export default (options) => {
	return {
		id: "naver",
		name: "Naver",
		type: "oauth",
		version: "2.0",
		params: { grant_type: "authorization_code" },
		profileUrl: "https://openapi.naver.com/v1/nid/me",
		accessTokenUrl: "https://nid.naver.com/oauth2.0/token",
		authorizationUrl: "https://nid.naver.com/oauth2.0/authorize?response_type=code",
		profile: (profile) => {
			return {
				id: profile.response.id,
				name: profile.response.name,
				email: profile.response.email,
				image: profile.response.image,
			};
		},
		...options,
	};
};
```

However, I met this problem. I think there must be something more I need to implement to add a new provider. Could you help me? Thanks a lot.

```
TypeError: next_auth_providers__WEBPACK_IMPORTED_MODULE_1___default.a.Naver is not a function
```

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [x] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by LoriKarikari** [comment]
You should also export the new provider here: https://github.com/nextauthjs/next-auth/blob/main/src/providers/index.js

**Comment by DominicFinn** [comment]
Hi there, the source code appears to have moved on but I have a similar request. I'd like to add try and add a provider for UNiDAYS which is a student network and verification provider. 

In the example by @heiwais25 the location of the providers is straight under `src/providers`. It looks like the repo has moved on since then. The providers appear to be in a package now but the package is also called next-auth.  

Should that package be edited directly in this repo? It doesn't seem like that would be the correct thing to do! 🙃
